### ISSUE TITLE
cURL vs RestSharp vs Rest.net behaving differently with Tumblr

### ISSUE THREAD

**Issue by thomasd3** [restsharp/RestSharp#1171]
I am trying to get data from the Tumblr API and get different results since they introduced the GDPR/Advertisers page in front. The page is supposed to appear once and set a cookie.

With cUrl:

    curl -i -v -X GET "https://nochillmemes.tumblr.com/api/read/json?type=photo&num=50&start=1"

I get proper results.

With Rest.Net:

    var Client = new RestClient("https://nochillmemes.tumblr.com");
    var Request	= new RestRequest("/api/read/json?filter=text&type=photo&num=50&start=1", Http.Method.GET);
    var R = Client.Execute(Request);

returns the GDPR page, but:

    var Client = new RestClient("https://nochillmemes.tumblr.com");
    var Request	= new RestRequest("/api/read/json?filter=text&type=photo&num=50&start=1", Http.Method.GET);
    Request.AddHeader("user-agent", "curl/7.54.0");
    var R = Client.Execute(Request);

returns the proper data! so tumblr returns different results based on the user agent.

With RestSharp:

    var Client = new RestClient("https://nochillmemes.tumblr.com");
    var Request	= new RestRequest("/api/read/json?filter=text&type=photo&num=50&start=1", Method.GET);
    var R = Client.Execute(Request);

returns the GDPR page, and:

    var Client = new RestClient("https://nochillmemes.tumblr.com");
    var Request	= new RestRequest("/api/read/json?filter=text&type=photo&num=50&start=1", Method.GET);
    Request.AddHeader("user-agent", "curl/7.54.0");
    var R = Client.Execute(Request);

still returns the GDPR page...

So, in summary:

- cURL gives me the API results
- Rest.Net gives me the API results only if it pretends to be cURL
- Restsharp always gives me the GDPR page

What could be the reason for this?

I'm trying to understand what is happening because I don't want to rely on something that works without knowing why :)

I posted about this on SO as well: https://stackoverflow.com/questions/52404588/curl-vs-restsharp-vs-rest-net-behaving-differently-with-tumblr

**Comment by zeldafreak** [comment]
I think this can be closed. It was answered on SO (use `Request.UserAgent` instead of `Request.AddHeader`): https://stackoverflow.com/a/52405248/399124

**Comment by thomasd3** [comment]
Before closing it, shouldn't this behavior be documented, or at least have a warning printed out if we set a user agent header? since the result will not match the expected behavior.

**Comment by alexeyzimarev** [comment]
Documented.
